### DISCUSSION TITLE
Create a kafka cluster with existing pv

### DISCUSSION THREAD

**Discussion by dearboll23** [strimzi/strimzi-kafka-operator#7086] [Q&A] [answered]
Hi all. I wanna to create a kafka cluster with existing pv (managed by dynamic provisioner), in case of wrongly deletion. I've tried using the storage.selector, but dynamic provisioner did'nt support selector (logged as "claim Selector is not supported"). So have any suggestions and best practice for preventing wrongly deletion? Any help is appreciated.

**Comment by scholzj** [comment, marked as answer]
I think you should be able to use the following procedure for recovering the cluster from the PVs: https://strimzi.io/docs/operators/latest/full/configuring.html#cluster-recovery-str

It is not clear if your PVs are empty or already have some data: 
* Keep in mind, that if they already have some data, they need to be in the _Strimzi format_ (i.e. the path must match what Strimzi is using etc.). 
* If they are empty, it should be fine, although I'm not sure why not just use Storage Class and have them dynamically provisioned in that case. But the procedure from the link should work if for some reason you want to go this way.

**Reply by dearboll23** [reply]
Thanks scholzj. The case is PVs already have some data and managed by Strimzi kafka operator. The guide is awesome but I wonder that I need to add 3 PVCs manually (and they will be bound with the PVs retained), then I create the kafka.yaml, kafka operator won't do anything coz the PVCs already existed?

**Reply by scholzj** [reply]
Yes, if the PVCs already exist, the operator should just take them over and use them.

**Reply by dearboll23** [reply]
Great thanks for the answer
